src/nvim/ex_docmd.c

```diff
@@ -7710,8 +7710,12 @@ void free_cd_dir(void)
 /// Deal with the side effects of changing the current directory.
 ///
 /// @param scope  Scope of the function call (global, tab or window).
-void post_chdir(CdScope scope, bool trigger_dirchanged)
+void post_chdir(CdScope scope, bool trigger_dirchanged, char_u *previous_dir)
 {
+  if (previous_dir != NULL) {
+    prev_dir = vim_strsave(previous_dir);
```

Comment:
I believe this is a leak, you need to free the old prev_dir.

https://github.com/neovim/neovim/blob/70b854dbd2ae9763f8061052e1074cb14be27c00/src/nvim/ex_docmd.c#L7787